utils: Add a helper script for creating platform checks

mrogers950 opened the pull request:
#### Description:

Add utils/add_platform_rule.py which can be used by content developers to bootstrap new platform check content. It contains subcommands to create a new rule and verify the check criteria both in-cluster with compliance-operator and locally with podman.

Use example:
```
 $ utils/add_platform_rule.py create --rule=ocp_proxy_has_ca --type="proxies.config" --name="cluster" --yamlpath=".spec.trustedCA.name" --match="[a-zA-Z0-9]*"
 creating check for "/apis/config.openshift.io/v1/proxies/cluster" with yamlpath ".spec.trustedCA.name" satisfying match of "[a-zA-Z0-9]*"
 wrote applications/openshift/ocp_proxy_has_ca/rule.yml
 wrote applications/openshift/ocp_proxy_has_ca/oval/shared.xml

 $ utils/add_platform_rule.py cluster-test --rule=ocp_proxy_has_ca
 testing rule ocp_proxy_has_ca in-cluster
 deploying compliance-operator
 pushing image build to cluster
 waiting for cleanup from previous test run
 output from last phase check: LAUNCHING NOT-AVAILABLE
 output from last phase check: RUNNING NOT-AVAILABLE
 output from last phase check: AGGREGATING NOT-AVAILABLE
 output from last phase check: DONE COMPLIANT
 COMPLIANT
```

#### Rationale:

Remove some of the guesswork in creating OCP4 content and make manual testing easier for new content developers.

pep8speaks commented:
Hello @mrogers950! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`utils/add_platform_rule.py`](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py):

> [Line 12:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L12): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (145 > 99 characters)
> [Line 13:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L13): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (136 > 99 characters)
> [Line 14:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L14): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (105 > 99 characters)
> [Line 16:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L16): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (158 > 99 characters)
> [Line 17:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L17): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (138 > 99 characters)
> [Line 35:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L35): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (171 > 99 characters)
> [Line 37:14](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L37): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 56:14](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L56): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 68:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L68): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (125 > 99 characters)
> [Line 72:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L72): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (120 > 99 characters)
> [Line 77:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L77): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (142 > 99 characters)
> [Line 84:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L84): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (136 > 99 characters)
> [Line 101:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L101): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (108 > 99 characters)
> [Line 106:17](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L106): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 118:19](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L118): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 126:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L126): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (111 > 99 characters)
> [Line 132:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L132): [E101](https://duckduckgo.com/?q=pep8%20E101) indentation contains mixed spaces and tabs
> [Line 132:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L132): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 133:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L133): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 134:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L134): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 135:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L135): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 136:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L136): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 138:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L138): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 139:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L139): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 140:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L140): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 141:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L141): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 143:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L143): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 144:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L144): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 145:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L145): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 145:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L145): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (144 > 99 characters)
> [Line 146:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L146): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 147:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L147): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 148:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L148): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 149:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L149): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 150:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L150): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 152:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L152): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 152:9](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L152): [E711](https://duckduckgo.com/?q=pep8%20E711) comparison to None should be 'if cond is None:'
> [Line 153:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L153): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 153:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L153): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (119 > 99 characters)
> [Line 154:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L154): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 156:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L156): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 156:79](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L156): [E225](https://duckduckgo.com/?q=pep8%20E225) missing whitespace around operator
> [Line 156:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L156): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (111 > 99 characters)
> [Line 157:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L157): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 158:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L158): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 159:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L159): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 160:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L160): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 162:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L162): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 163:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L163): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 164:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L164): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 165:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L165): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 166:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L166): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 167:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L167): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 167:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L167): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (146 > 99 characters)
> [Line 168:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L168): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 170:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L170): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 174:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L174): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 175:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L175): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 176:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L176): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 178:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L178): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 180:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L180): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 181:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L181): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 183:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L183): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 184:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L184): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 185:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L185): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 187:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L187): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 188:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L188): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 189:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L189): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 191:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L191): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 192:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L192): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 193:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L193): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 195:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L195): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 196:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L196): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 197:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L197): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 198:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L198): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 199:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L199): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 200:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L200): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 201:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L201): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 202:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L202): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 203:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L203): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 204:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L204): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 206:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L206): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 207:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L207): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 208:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L208): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 209:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L209): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 210:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L210): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 212:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L212): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 213:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L213): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 214:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L214): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 215:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L215): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 216:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L216): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 217:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L217): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 218:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L218): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 219:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L219): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 220:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L220): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 221:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L221): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 222:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L222): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 223:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L223): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 225:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L225): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 226:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L226): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 227:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L227): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 228:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L228): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 229:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L229): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 229:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L229): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (129 > 99 characters)
> [Line 230:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L230): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 231:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L231): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 232:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L232): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 233:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L233): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 234:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L234): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 235:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L235): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 236:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L236): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 237:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L237): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 239:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L239): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 239:17](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L239): [E711](https://duckduckgo.com/?q=pep8%20E711) comparison to None should be 'if cond is None:'
> [Line 240:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L240): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 241:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L241): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 243:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L243): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 244:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L244): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 248:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L248): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 249:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L249): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 250:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L250): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 252:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L252): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 253:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L253): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 253:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L253): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (163 > 99 characters)
> [Line 254:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L254): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 258:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L258): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (3)
> [Line 259:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L259): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 259:29](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L259): [E711](https://duckduckgo.com/?q=pep8%20E711) comparison to None should be 'if cond is None:'
> [Line 260:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L260): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 261:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L261): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 263:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L263): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 263:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L263): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (134 > 99 characters)
> [Line 264:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L264): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 265:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L265): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 265:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L265): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (133 > 99 characters)
> [Line 266:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L266): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 267:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L267): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 267:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L267): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (104 > 99 characters)
> [Line 268:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L268): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 268:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L268): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (108 > 99 characters)
> [Line 269:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L269): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 269:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L269): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (111 > 99 characters)
> [Line 270:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L270): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 270:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L270): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (118 > 99 characters)
> [Line 271:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L271): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 271:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L271): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (141 > 99 characters)
> [Line 272:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L272): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 273:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L273): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 273:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L273): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (154 > 99 characters)
> [Line 274:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L274): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 274:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L274): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (116 > 99 characters)
> [Line 275:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L275): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 275:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L275): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (114 > 99 characters)
> [Line 276:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L276): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 278:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L278): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 278:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L278): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (135 > 99 characters)
> [Line 279:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L279): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 280:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L280): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 281:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L281): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 283:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L283): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 283:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L283): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (151 > 99 characters)
> [Line 284:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L284): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 285:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L285): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 285:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L285): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (124 > 99 characters)
> [Line 286:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L286): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 286:100](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L286): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (131 > 99 characters)
> [Line 287:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L287): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 289:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L289): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 290:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L290): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 294:1](https://github.com/ComplianceAsCode/content/blob/cd3dfacd03b9774b684c1c87ed40e9a097413f33/utils/add_platform_rule.py#L294): [E101](https://duckduckgo.com/?q=pep8%20E101) indentation contains mixed spaces and tabs

mildas commented:
**Changes identified:**
Others:
 Python file utils/add_platform_rule.py is newly added.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)

mildas commented:
**Changes identified:**
Others:
 Python file utils/add_platform_rule.py is newly added.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)

mildas commented:
**Changes identified:**
Others:
 Python file utils/add_platform_rule.py is newly added.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)

mrogers950 commented:
@jhrozek @JAORMX PTAL.

JAORMX commented:
/ok-to-test

JAORMX commented:
/test all

jhrozek commented on utils/add_platform_rule.py:
```diff
@@ -0,0 +1,321 @@
+rationale: TBD
+
+identifiers:
+    cce@ocp4: 84209-6
```
What does this identifier mean? Is it OK for all generated rules to have the same?

mrogers950 replied:
Oh right, It's the unique CCE id for the rule. So each rule needs one from shared/references/cce-redhat-avail.txt . When you add a rule you have to cut from that file. I added a flag to be able to fill in the reference, but leaving the value and removal from cce-redhat-avail.txt up to the user for now..

jhrozek commented on utils/add_platform_rule.py:
```diff
@@ -0,0 +1,321 @@
+identifiers:
+    cce@ocp4: 84209-6
+
+severity: medium
```
Maybe the severity should be TBD to force the creator to set it

jhrozek commented on utils/add_platform_rule.py:
```diff
@@ -0,0 +1,321 @@
+
+{{{{% if target_oval_version >= YAML_TEST_OVAL_VERSION %}}}}
+<def-group>
+      <definition class="compliance" version="1" id="{{{{{{ rule_id }}}}}}">
```
I wonder if this could be an OVAL macro, there are already several under `shared/`

mrogers950 replied:
We might be able to macrotize but I'd rather wait until we get some more rules created and find the common need macros

jhrozek commented on utils/add_platform_rule.py:
```diff
@@ -0,0 +1,321 @@
+    test_parser.add_argument('--rule', required=True,
+                             help='The name of the rule to test.')
+    test_parser.add_argument(
+        '--file', required=True, default="build/ssg-ocp4-ds.xml", help='The path to the datastream file.')
```
If there's a default, the argument doesn't have to be required, right?

jhrozek commented on utils/add_platform_rule.py:
```diff
@@ -0,0 +1,321 @@
+
+    print('testing rule %s locally' % args.rule)
+    ret_code, out = subprocess.getstatusoutput(
+        'podman run -it -v %s:/content/:z -v %s:/tmp/:z %s %s' % (args.file, args.objectdir, OSCAP_TEST_IMAGE, OSCAP_CMD_OPTS))
```
I'm having issues with this command. It seems like the objectdir is actually required, the default `/tmp` doesn't work for me, I'm getting `Error: relabel failed "/tmp": SELinux relabeling of /tmp is not allowed`. Also, just using the file as the volume doesn't work either, I'm getting errors like `Error: error creating named volume "build/ssg-ocp4-ds.xml": error running volume create option: names must match [a-zA-Z0-9][a-zA-Z0-9_.-]*: invalid argument`. Did you maybe want to share the parent directory of the DS file instead?

jhrozek replied:
Also the test profile doesn't exist by default..

jhrozek commented on utils/add_platform_rule.py:
```diff
@@ -0,0 +1,321 @@
+        <description>{DESC}</description>
+      </metadata>
+      <criteria operator="AND">
+        <criterion comment="{DESC}" negate="{NEGATE}" test_ref="test_{{{{{{ rule_id }}}}}}"/>
```
This is causing the build to fail, because the Python boolean gets converted to True/False. My crude fix was:
```
@@ -168,7 +169,7 @@ def createFunc(args):
     print('wrote ' + rule_yaml_path)
     with open(shared_xml_path, 'w') as file:
         file.write(OVAL_TEMPLATE.format(TITLE=args.title, DESC=args.description,
-                                        URL=url, YAMLPATH=args.yamlpath, MATCH=args.match, NEGATE=args.negate))
+                                        URL=url, YAMLPATH=args.yamlpath, MATCH=args.match, NEGATE=str(args.negate).lower()))
```

mildas commented:
**Changes identified:**
Others:
 Python file utils/add_platform_rule.py is newly added.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)

mildas commented:
**Changes identified:**
Others:
 Python file utils/add_platform_rule.py is newly added.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)

mrogers950 commented:
@JAORMX @jhrozek thanks for the reviews! The test command was indeed bogus, I've fixed it up and added a use example to the description text. Other comments are addressed as well.

JAORMX commented on utils/add_platform_rule.py:
```diff
@@ -0,0 +1,383 @@
+rationale: TBD
+
+identifiers:
+    {IDENT}
```
This wouldn't be a valid type for identifiers. a dictionary of identifiers is what's really expected here. IMO you should just leave this as an empty dict in yaml, and let the user edit it later. If not, somehow you'd need to parse the dict of identifiers.

jhrozek commented on utils/add_platform_rule.py:
```diff
@@ -0,0 +1,383 @@
+            print('build failed: %s' % out)
+            return 1
+
+    pod_cmd = 'podman run -it --security-opt label=disable -v "%s:/content" -v "%s:/tmp" %s %s' % (args.file, args.objectdir, OSCAP_TEST_IMAGE, OSCAP_CMD_OPTS)
```
The test doesn't work for me, I'm getting:
```
utils/add_platform_rule.py test --rule=ocp_proxy_has_ca
testing rule ocp_proxy_has_ca locally
Traceback (most recent call last):
  File "utils/add_platform_rule.py", line 383, in <module>
    sys.exit(main())
  File "utils/add_platform_rule.py", line 379, in main
    return args.func(args)
  File "utils/add_platform_rule.py", line 313, in testFunc
    pod_cmd = 'podman run -it --security-opt label=disable -v "%s:/content" -v "%s:/tmp" %s %s' % (args.file, args.objectdir, OSCAP_TEST_IMAGE, OSCAP_CMD_OPTS)

```

mildas commented:
**Changes identified:**
Others:
 Python file utils/add_platform_rule.py is newly added.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)

jhrozek commented:
Thank you this works for me. I have one more suggestion, but it's really just a suggestion, so feel free to ignore it or make use of it in a subsequent PR: I was wondering if it makes sense to add `--verbose INFO` after the `xccdf eval` because this is a devel tool and chances are you want to see some more verbose output. Or even provide an optional flag for the `test` command that sets the verbosity if needed.

But otherwise LGTM, really nice work.

jhrozek commented:
/test all

mildas commented:
**Changes identified:**
Others:
 Python file utils/add_platform_rule.py is newly added.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)

mildas commented:
**Changes identified:**
Others:
 Python file utils/add_platform_rule.py is newly added.

**Recommended tests to execute:**
 (cd build && cmake ../ && ctest -j4)

mrogers950 commented:
@jhrozek @redhatrises thanks for the reviews, I've updated this and also fixed a bug with the flags to skip building/pushing for quicker re-tests. They are now --skip-deploy and --skip-build.
